test/infrastructure/docker/docker/machine.go

```diff
@@ -430,3 +434,21 @@ func (m *Machine) machineImage(version *string) string {
 
 	return fmt.Sprintf("%s:%s", defaultImageName, versionString)
 }
+
+func logContainerDebugInfo(ctx context.Context, name string) {
```

Comment:
xref: https://github.com/kubernetes-sigs/cluster-api/pull/4416#discussion_r604939121

@fabriziopandini   Is the pattern in general to just pass the context and always get the logger from there? (in cases where we have to pass the context anyway, which is probably almost everywhere)

Reply:
Yes, this is the pattern we are adopting after controller-runtime moved the logger into the context

Reply:
Okay thx.